File: ticketdesk-backend/src/main/java/com/ticketdesk/service/TicketService.java
package com.ticketdesk.service;

import com.ticketdesk.dto.PagedResponse;
import com.ticketdesk.dto.TicketRequest;
import com.ticketdesk.dto.TicketResponse;
import com.ticketdesk.entity.Category;
import com.ticketdesk.entity.Priority;
import com.ticketdesk.entity.Status;
import com.ticketdesk.security.UserPrincipal;

public interface TicketService {
    TicketResponse createTicket(TicketRequest request, UserPrincipal currentUser);
    
    PagedResponse<TicketResponse> getTickets(
            Status status,
            Priority priority,
            Category category,
            Long createdBy,
            String search,
            int page,
            int size,
            String sortBy,
            String sortDir,
            UserPrincipal currentUser
    );

    TicketResponse getTicketById(Long id, UserPrincipal currentUser);

    TicketResponse updateTicket(Long id, TicketRequest request, UserPrincipal currentUser);

    void deleteTicket(Long id, UserPrincipal currentUser);
}

File: ticketdesk-frontend/src/services/ticketService.js
import api from './api';

export const ticketService = {
  createTicket: async (ticketData) => {
    const response = await api.post('/tickets', ticketData);
    return response.data;
  },

  getTickets: async (params = {}) => {
    const response = await api.get('/tickets', { params });
    return response.data;
  },

  getTicketById: async (id) => {
    const response = await api.get(`/tickets/${id}`);
    return response.data;
  },

  updateTicket: async (id, ticketData) => {
    const response = await api.put(`/tickets/${id}`, ticketData);
    return response.data;
  },

  deleteTicket: async (id) => {
    await api.delete(`/tickets/${id}`);
  },
};
